static/app/components/events/interfaces/spans/spanTreeModel.tsx

```diff
@@ -305,57 +319,219 @@ class SpanTreeModel {
+                    ? toggleNestedSpanGroup === undefined
+                      ? this.isNestedSpanGroupExpanded
+                      : isNestedSpanGroupExpanded
+                    : false,
+                  addTraceBounds,
+                  removeTraceBounds,
+                  isAutogroupSiblingFeatureEnabled,
+                })
+              );
+
+              acc.previousSiblingEndTimestamp = spanModel.span.timestamp;
+            });
+
+            return acc;
+          }
+
+          // NOTE: I am making the assumption here that grouped sibling spans will not have children.
+          // By making this assumption, I can immediately wrap the grouped spans here without having
+          // to recursively traverse them.
+
+          // This may not be the case, and needs to be looked into later
+
+          const wrappedSiblings: EnhancedSpan[] = group.map((spanModel, index) => {
+            const enhancedSibling: EnhancedSpan = {
+              type: 'span',
+              span: spanModel.span,
+              numOfSpanChildren: 0,
+              treeDepth: treeDepth + 1,
+              isLastSibling:
+                index === group.length - 1 &&
+                groupIndex === groupedDescendants.length - 1,
+              isFirstSiblingOfGroup: index === 0,
+              continuingTreeDepths: descendantContinuingTreeDepths,
+              fetchEmbeddedChildrenState: spanModel.fetchEmbeddedChildrenState,
+              showEmbeddedChildren: spanModel.showEmbeddedChildren,
+              toggleEmbeddedChildren: spanModel.toggleEmbeddedChildren({
+                addTraceBounds,
+                removeTraceBounds,
+              }),
+              toggleNestedSpanGroup: undefined,
+              toggleSiblingSpanGroup:
+                index === 0 ? this.toggleSiblingSpanGroup : undefined,
+            };
+
+            return enhancedSibling;
+          });
+
+          if (this.isSpanFilteredOut(props, group[0])) {
+            group.forEach(spanModel =>
+              acc.descendants.push({
+                type: 'filtered_out',
+                span: spanModel.span,
+              })
+            );
+            return acc;
+          }
+
+          const groupedSiblingsSpan: EnhancedProcessedSpanType = {
+            type: 'span_group_sibling',
+            span: this.span,
+            treeDepth: treeDepth + 1,
+            continuingTreeDepths,
+            spanSiblingGrouping: wrappedSiblings,
+            isLastSibling: groupIndex === groupedDescendants.length - 1,
+            toggleSiblingSpanGroup: this.toggleSiblingSpanGroup,
+          };
+
+          acc.previousSiblingEndTimestamp =
+            wrappedSiblings[wrappedSiblings.length - 1].span.timestamp;
+
+          // Check if the group is currently expanded or not
+          const key = `${group[0].span.op}.${group[0].span.description}`;
+          if (this.expandedSiblingGroups.has(key)) {
+            acc.descendants.push(...wrappedSiblings);
```

Comment:
@0Calories Since you've avoided using `span.getSpansList()` on each span in the expanded sibling group, you'll need to `filter()` for each `span` in `wrappedSiblings` with `this.isSpanFilteredOut(props, span)`. 

This due to the fact that `this.props.filterSpans.spanIDs` will still need to be considered.